src/tasks/Tasks.ts

```diff
@@ -0,0 +1,869 @@
+    // in which case don't do anything, and the lock remains unlocked
+    if (this.queuingLockReleaser != null) {
+      this.queuingLockReleaser();
+    }
+  }
+
+  protected async queueTask(
+    taskId: TaskId,
+    taskScheduleTime: number,
+    taskPriority: TaskPriority,
+  ): Promise<void> {
+    await this.db.withTransactionF(async (tran) => {
+      this.schedulerLogger.debug(`Queuing Task ${taskId.toMultibase('base32hex')}`);
+      // Put task into the queue index
+      await tran.put(
+        [
+          ...this.tasksQueuedDbPath,
+          utils.lexiPackBuffer(taskPriority),
+          utils.lexiPackBuffer(taskScheduleTime),
+          taskId.toBuffer(),
+        ],
+        taskId.toBuffer(),
+        true
+      );
+      tran.queueSuccess(() => {
+        this.schedulerLogger.debug(`Queued Task ${taskId.toMultibase('base32hex')}`);
+        this.triggerQueuing();
+      });
+    });
+  }
+
+  protected async startTask(
+    taskId: TaskId,
+    taskData: TaskData,
+    tran?: DBTransaction,
+  ): Promise<void> {
+    if (tran == null) {
+      return this.db.withTransactionF((tran) => this.startTask(taskId, taskData, tran));
+    }
+    const taskIdString = taskId.toString() as TaskIdString;
+    const taskHandler = this.getHandler(taskData.handlerId);
+    if (taskHandler == null) {
+      this.queueLogger.debug(`Failed Task ${taskId.toMultibase('base32hex')} - No Handler Registered`);
+      this.taskEvents.dispatchEvent(
+        new TaskEvent(
+          taskIdString,
+          {
+            detail: {
+              status: 'failure',
+              reason: new tasksErrors.ErrorSchedulerHandlerMissing()
+            }
+          }
+        )
+      );
+      await this.gcTask(taskId, tran);
+      return;
+    }
+    // Put task into the active index
+    // this index will be used to retry tasks if they don't finish
+    await tran.put(
+      [
+        ...this.tasksActiveDbPath,
+        taskId.toBuffer()
+      ],
+      taskId.toBuffer(),
+      true
+    );
+    tran.queueSuccess(() => {
+
+      // Dummy values for now
+      const timer = new Timer;
+      const abortController = new AbortController();
+
+      const taskP = taskHandler(
+        ...taskData.parameters,
+        {
+          timer: timer,
+          signal: abortController.signal
+        }
+      ).then((result: any) => {
+        this.queueLogger.debug(`Succeeded Task ${taskId.toMultibase('base32hex')}`);
+        // If no event listeners, then only side effects are recorded
+        this.taskEvents.dispatchEvent(
+          new TaskEvent(
+            taskIdString,
+            {
+              detail: {
+                status: 'success',
+                result
+              }
+            }
+          )
+        );
+      }, (reason: any) => {
+        this.queueLogger.debug(`Failed Task ${taskId.toMultibase('base32hex')} - Reason: ${reason}`);
+        this.taskEvents.dispatchEvent(
+          new TaskEvent(
+            taskIdString,
```

Comment:
I think the 'Failed Task...' should be a warning.

Reply:
Yea, that can be a warning. Same with the other error message about the task handler ID not being registered. That should be an `this.queueLogger.error` message though.

Reply:
No handler is now `error`, and failed task is just `warn`.